android/app/src/test/java/org/smartregister/fhircore/data/SamplePatientsTest.kt

```diff
@@ -0,0 +1,194 @@
+import java.util.Date
+import org.hl7.fhir.r4.model.ContactPoint
+import org.hl7.fhir.r4.model.Enumerations
+import org.hl7.fhir.r4.model.HumanName
+import org.hl7.fhir.r4.model.Patient
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.smartregister.fhircore.viewmodel.PatientListViewModel
+
+class SamplePatientsTest {
+
+  private lateinit var samplePatients: SamplePatients
+
+  @Before
+  fun setUp() {
+    samplePatients = SamplePatients()
+  }
+
+  @Test
+  fun testGetPatientItemsShouldReturnNormalizePatientItemList() {
+    val patients = getPatients()
+    val items = samplePatients.getPatientItems(patients)
+
+    verifyPatient(patients[0], items[0])
+    verifyPatient(patients[1], items[1])
+  }
+
+  @Test
+  fun testGetObservationItemsShouldReturnNormalizeObservationItemList() {
+    val observations = samplePatients.getObservationItems(getObservationJson())
+
+    Assert.assertEquals(1, observations.size)
+    Assert.assertEquals("1", observations[0].id)
+    Assert.assertEquals("Carbon Dioxide", observations[0].code)
+    Assert.assertEquals("2002-07-28T15:08:13-05:00", observations[0].effective)
+    Assert.assertEquals("28.85 mmol/L", observations[0].value)
+  }
+
+  private fun verifyPatient(patient: Patient, item: PatientListViewModel.PatientItem) {
+    Assert.assertEquals(patient.id, item.logicalId)
+    Assert.assertEquals(patient.name[0].nameAsSingleString, item.name)
+    Assert.assertEquals(patient.genderElement.valueAsString, item.gender)
+    Assert.assertEquals(patient.birthDateElement.valueAsString, item.dob)
+    Assert.assertEquals(patient.telecom[0].value, item.phone)
+    Assert.assertEquals(patient.text.div.valueAsString, item.html)
+  }
+
+  private fun getPatients(): List<Patient> {
+    return listOf(
+
+      // patient 1
+      Patient().apply {
+        id = "patient_1"
+        gender = Enumerations.AdministrativeGender.MALE
+        name = getPatientName("mc", "jane")
+        telecom = getTelecom("123")
+        birthDate = getBirthDate("2000-02-03")
+        text.divAsString = "a"
+      },
+
+      // patient 2
+      Patient().apply {
+        id = "patient_2"
+        gender = Enumerations.AdministrativeGender.FEMALE
+        name = getPatientName("julie", "sam")
+        telecom = getTelecom("456")
+        birthDate = getBirthDate("2003-04-13")
+        text.divAsString = "b"
+      }
+    )
+  }
+
+  private fun getPatientName(given: String, family: String): MutableList<HumanName> {
+    return mutableListOf(
+      HumanName().apply {
+        addGiven(given)
+        setFamily(family)
+      }
+    )
+  }
+
+  private fun getTelecom(telecom: String): MutableList<ContactPoint> {
+    return mutableListOf(ContactPoint().apply { value = telecom })
+  }
+
+  private fun getBirthDate(birthDate: String): Date? {
+    return SimpleDateFormat("yyyy-MM-dd").parse(birthDate)
+  }
+
+  private fun getObservationJson(): String {
+    return """
+      {
```

Comment:
I don't know if this should go in a flat file in an test assets folder. I think it's OK here as is if we don't ever plan to use it elsewhere, ie we don't want to repeat it in code, in that case we should refactor to a flat file

Reply:
I thought it's good for now but we can easily refactor it in the future.